docs/configuredgraphfactory.txt

```diff
@@ -0,0 +1,504 @@
+
+In particular, the `JanusGraphManager` provides
+1. a coordinated mechanism by which to instantiate graph references on a given JanusGraph node
+2. a graph reference tracker (or cache)
+
+Any graph you create using the "graph.graphname" property will go through the `JanusGraphManager` and thus be instantiated in a coordinated fashion. The graph reference will also be placed in the graph cache on the JVM in question.
+
+Thus, any graph you open using the "graph.graphname" property that has already been instantiated on the JVM in question will be retrieved from the graph cache.
+
+This is why updates to your configurations require a few steps to guarantee correctness.
+
+[[usingthejanusgraphmanager]]
+How To Use The JanusGraphManager
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+
+This is a new configuration option you can use when defining a property in your configuration that defines how to access a graph. All configurations that include this property will result in the graph instantiation happening through the `JanusGraphManager` (process explained above).
+
+For backwards compatability, any graphs that do not supply this parameter but supplied at server start in your graphs {} object in your .yaml file, these graphs will be bound through the JanusGraphManager denoted by their `key` supplied for that graph. For example, if your .yaml graphs object looks like:
+
+[source, properties]
+----
+graphs \{
+graph1: conf/graph1.properties,
+graph2: conf/graph2.properties
+}
+----
+
+but `conf/graph1.properties` and `conf/graph2.properties` do not include the property `graph.graphname`, then these graphs will be stored in the JanusGraphManager and thus bound in your gremlin script executions as `graph1` and `graph2`, respectively.
+
+
+[[important]]
+Important
+^^^^^^^^^
+
+For convenience, if your configuration used to open a graph specifies "graph.graphname", but does not specify the backend's storage directory, tablename, or keyspacename, then the relevant parameter will automatically be set to the value of "graph.graphname". However, if you supply one of those parameters, that value will always take precedence. And if you supply neither, they default to the configuration option's default value.
+
+One special case is `storage.root` configuration option. This is a new configuration option used to specify the base of the directory that will be used for any backend requiring local storage directory access. If you supply this parameter, you must also supply the "graph.graphname" property, and the absolute storage directory will be equal to the value of the "graph.graphname" property appended to the value of the "storage.root" property.
+
+Below are some example use cases:
+
+1. Create a template configuration for my Cassandra backend such that each graph created using this configuration gets a unique keyspace equivalent to the `String` <graphName> provided to the factory:
```

Comment:
It looks like the Asciidoc is getting confused by the code blocks in between numbered bullets. Try using `1)` instead of `1.` or just leave out the numbering.